gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthServer.js

```diff
@@ -66,19 +66,39 @@ gpii.oauth2.oauth2orizeServer.listenOauth2orize = function (oauth2orizeServer, c
     });
 
     oauth2orizeServer.deserializeClient(function (id, done) {
-        return done(null, clientService.getClientById(id));
+        var clientPromise = clientService.getClientById(id);
+        clientPromise.then(function (client) {
+            return done(null, client);
+        });
     });
 
     oauth2orizeServer.grant(oauth2orize.grant.code(function (client, redirectUri, user, ares, done) {
-        return done(null, authorizationService.grantAuthorizationCode(user.id, client.id, redirectUri, ares.selectedPreferences));
+        // TODO: Update the user interface to support multiple tokens
+        // per user rather than using a single default
+        var gpiiToken = user.defaultGpiiToken;
+        var authPromise = authorizationService.grantAuthorizationCode(gpiiToken, client.id, redirectUri, ares.selectedPreferences);
+
+        authPromise.then(function (auth) {
+            return done(null, auth);
+        });
     }));
 
     oauth2orizeServer.exchange(oauth2orize.exchange.code(function (client, code, redirectUri, done) {
-        return done(null, authorizationService.exchangeCodeForAccessToken(code, client.id, redirectUri));
+        var authPromise = authorizationService.exchangeCodeForAccessToken(code, client.id, redirectUri);
+        authPromise.then(function (code) {
```

Comment:
These should all be handled by a common utility which we might name gpii.oauth2.oauth2orize.promiseToDone which generally gears the bizarre oauth2orize "done" callback to any promise